### PR TITLE
Fix types for Paginator

### PR DESCRIPTION
**Pull request by derrabus**

### CODE REVIEW

**Comment by greg0ire** [comment]
If this is a fix, why target 2.14.x?

File: lib/Doctrine/ORM/Tools/Pagination/Paginator.php [hunk 1]
```diff
@@ -27,6 +27,7 @@
  * The paginator can handle various complex scenarios with DQL.
  *
  * @template-covariant T
+ * @implements IteratorAggregate<array-key, T>
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Tools/Pagination/Paginator.php:29]: What does this fix?

Comment (greg0ire) [reply lib/Doctrine/ORM/Tools/Pagination/Paginator.php:29]: Ah I think I know: it fixes a warning about implementing IteratorAggregate w/o specifying the types. I have that same fix in a branch that fixes part of the issues detected by Psalm v5, I will share it soon.

Comment (derrabus) [reply lib/Doctrine/ORM/Tools/Pagination/Paginator.php:29]: Yes. I found it as well while playing with Psalm 5.

**Comment by derrabus** [comment]
> If this is a fix, why target 2.14.x?

No strong opinion here, I can rebase onto 2.13.x.

**Comment by derrabus** [comment]
Done.

**Review by greg0ire — APPROVED** [review]

**Pull request merged**